src/core/thread/mle_router.cpp

```diff
@@ -322,15 +320,11 @@ ThreadError MleRouter::HandleChildStart(otMleAttachFilter aFilter)
 
     if (mDeviceMode & ModeTlv::kModeFFD)
     {
-        advertiseDelay = Timer::SecToMsec(kReedAdvertiseInterval +
-                                          (otPlatRandomGet() % kReedAdvertiseJitter));
-        mAdvertiseTimer.Start(advertiseDelay);
+        mAdvertiseTimer.Start(
```

Comment:
I agree it would be nice to avoid unnecessary overhead.  I'm fine with using TrickleTimer as long as REEDs always set the new timer to fire at `REED_ADVERTISEMENT_INTERVAL + rand(0, REED_ADVERTISEMENT_MAX_JITTER)`.